lib/tests/helpers.test.js

```diff
@@ -216,4 +218,50 @@ describe("Submit and Template helpers", () => {
     expect(mockedFormikBag.setStatus).toBeCalledTimes(1);
     expect(mockedFormikBag.setStatus).toBeCalledWith("Error");
   });
+
+  test("Should submit with the correct csrf token", async () => {
+    getCsrfToken.mockReturnValueOnce("csrfTokenSecretValue");
+
+    mockedAxios.mockResolvedValue({
+      status: 200,
+      data: { received: true },
+    });
+    const mockedRouter = {
+      push: jest.fn(() => {}),
+    };
+    jest.spyOn(logMessage, "error");
+    await submitToAPI(mockResponses, {
+      props: {
+        formConfig: TestForm,
+        language: "en",
+        router: mockedRouter,
+      },
+    });
+    expect(mockedAxios.mock.calls[0][0].headers["X-CSRF-Token"]).toBe("csrfTokenSecretValue");
+    expect(mockedAxios.mock.calls[0][0].data).toEqual(buildFormDataObject(TestForm, mockResponses));
+    expect(mockedRouter.push).toBeCalledTimes(1);
+  });
+
+  test("Should fail on submit", async () => {
```

Comment:
This type of test should probably be extracted to it's own separate test file for the csrfProtected middleware.
The csrfProtected middleware tests should cover:
- Middleware passes (correct token and allowed methods)
- Middleware fails (incorrect token and unallowed methods)
- Middleware returns correct errors (can be incorporated in Middleware Fails tests to check for correct errors being thrown)